bin/packages/watch.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+const { execSync } = require( 'child_process' );
+const path = require( 'path' );
+const chalk = require( 'chalk' );
+
+/**
+ * Internal dependencies
+ */
+const getPackages = require( './get-packages' );
+
+const BUILD_CMD = `node ${ path.resolve( __dirname, './build.js' ) }`;
+
+let filesToBuild = new Map();
+
+const exists = ( filename ) => {
+	try {
+		return fs.statSync( filename ).isFile();
+	} catch ( e ) {}
+	return false;
+};
+const rebuild = ( filename ) => filesToBuild.set( filename, true );
+
+getPackages().forEach( ( p ) => {
+	const srcDir = path.resolve( p, 'src' );
+	try {
+		fs.accessSync( srcDir, fs.F_OK );
+		fs.watch( path.resolve( p, 'src' ), { recursive: true }, ( event, filename ) => {
+			const filePath = path.resolve( srcDir, filename );
+
+			if ( ( event === 'change' || event === 'rename' ) && exists( filePath ) ) {
+				// eslint-disable-next-line no-console
+				console.log( chalk.green( '->' ), `${ event }: ${ filename }` );
+				rebuild( filePath );
+			} else {
+				const buildFile = path.resolve( srcDir, '..', 'build', filename );
+				try {
+					fs.unlinkSync( buildFile );
+					process.stdout.write(
+						chalk.red( '  \u2022 ' ) +
+              path.relative( path.resolve( srcDir, '..', '..' ), buildFile ) +
+              ' (deleted)' +
+              '\n'
+					);
+				} catch ( e ) {}
+			}
+		} );
+	} catch ( e ) {
+		// doesn't exist
+	}
+} );
+
+setInterval( () => {
```

Comment:
Might be worth considering replacing this with a debounced function that's triggered in `rebuild`. Not really a major thing though.